### DISCUSSION TITLE
How to exclude optional dependencies when resolving Maven artifacts?

### DISCUSSION THREAD

**Discussion by guw** [eclipse-pde/eclipse.pde#1051] [Q&A] [answered]
Is there a recipe for excluding only optional compile dependencies? 

For example, given [this artifact ](https://mvnrepository.com/artifact/org.apache.commons/commons-configuration2/2.9.0) I only want the compile dependencies resolved in my TP, which are not optional.

**Comment by laeubi** [comment, marked as answer]
Maven artifacts are part of m2e not pde... but there is no way to exclude "optional" per se but you can exclude individual artifacts.

![grafik](https://user-images.githubusercontent.com/1331477/208263554-132f07e7-7a3b-4704-ba65-07610c4ccf6b.png)
![grafik](https://user-images.githubusercontent.com/1331477/208263589-84259fac-cef3-44b5-b125-a71b391d86fd.png)

**Reply by guw** [reply]
Thanks! That should work.
